notifications_utils/recipient_validation/phone_number.py

```diff
@@ -210,7 +84,17 @@ def _raise_if_service_cannot_send_to_uk_landline_but_tries_to(
 
     def validate(self, allow_international_number: bool = False, allow_uk_landline: bool = False) -> None:
         self._raise_if_service_cannot_send_to_international_but_tries_to(allow_international=allow_international_number)
-        self._raise_if_service_cannot_send_to_uk_landline_but_tries_to(allow_uk_landline=allow_uk_landline)
+        try:
+            self._raise_if_service_cannot_send_to_uk_landline_but_tries_to(allow_uk_landline=allow_uk_landline)
+        except InvalidPhoneError:
```

Comment:
what's the use case when we need to catch and do this extra normalisation? Shouldn't the phone number be already normalised from when we initialise the class?

Reply:
I'll add the following in a commit message:

"Adds code to handle cases where an international number is incorrectly parsed as a UK number the service doesn't have permission to send to.

There are certain cases where a valid international number is accidentally parsed as a UK landline. In these cases we want to try validating the number, but explicitly as an international number. If this fails, we want to turn the number back into a UK number and raise the correct validation error."

Without this code `tests/test_recipient_csv.py::test_international_recipients` because it thinks that `2304031000` (a valid Mauritius number) is actually meant to be a UK landline (this parses correctly) and then validation fails because the service cannot send to a UK landline. If this gets parsed as `+2304031000` it is correctly parsed as a Mauritus number and the test passes. As far as I'm aware this is an edge case that we want to support, and we can support it without causing other tests to fail.

So the logic goes:

- throw an exception if this is a landline and the service can't send to landlines
- catch the exception and make sure this isn't actually an international number we've mistaken for a landline
- if the error still throws, assume it was meant to be a landline and throw the original exception

Reply:
Ahh, that makes sense, thank you!